sdk/formrecognizer/azure-ai-formrecognizer/src/test/java/com/azure/ai/formrecognizer/FormTrainingClientTestBase.java

```diff
@@ -124,72 +95,72 @@ private static void validateErrorData(List<ErrorInformation> expectedErrors,
         }
     }
 
-    static void validateCustomModel(CustomFormModel expectedModel, CustomFormModel actualCustomModel) {
-        assertNotNull(actualCustomModel.getModelId());
-        assertEquals(expectedModel.getModelStatus(), actualCustomModel.getModelStatus());
-        validateErrors(expectedModel.getModelError(), actualCustomModel.getModelError());
-        assertNotNull(actualCustomModel.getCreatedOn());
-        assertNotNull(actualCustomModel.getLastUpdatedOn());
-        validateSubModels(expectedModel.getSubModels(), actualCustomModel.getSubModels());
-        validateTrainingDocuments(expectedModel.getTrainingDocuments(), actualCustomModel.getTrainingDocuments());
-    }
-
     static void validateAccountProperties(AccountProperties expectedAccountProperties,
         AccountProperties actualAccountProperties) {
         assertEquals(expectedAccountProperties.getLimit(), actualAccountProperties.getLimit());
         assertNotNull(actualAccountProperties.getCount());
     }
 
-    private static void validateTrainingDocuments(List<TrainingDocumentInfo> expectedTrainingDocuments,
-        List<TrainingDocumentInfo> actualTrainingDocuments) {
-        List<TrainingDocumentInfo> actualTrainingList = new ArrayList<>(actualTrainingDocuments);
-        List<TrainingDocumentInfo> expectedTrainingList =
-            new ArrayList<>(expectedTrainingDocuments);
-        assertEquals(expectedTrainingList.size(), actualTrainingList.size());
-        for (int i = 0; i < actualTrainingList.size(); i++) {
-            TrainingDocumentInfo expectedTrainingDocument = expectedTrainingList.get(i);
-            TrainingDocumentInfo actualTrainingDocument = actualTrainingList.get(i);
-            assertEquals(expectedTrainingDocument.getName(), actualTrainingDocument.getName());
-            assertEquals(expectedTrainingDocument.getPageCount(), actualTrainingDocument.getPageCount());
-            assertEquals(expectedTrainingDocument.getTrainingStatus(), actualTrainingDocument.getTrainingStatus());
-            validateErrors(expectedTrainingDocument.getDocumentErrors(), actualTrainingDocument.getDocumentErrors());
+    /**
+     * Deserialize test data from service.
+     *
+     * @return the deserialized raw response test data
+     */
+    static <T> T deserializeRawResponse(SerializerAdapter serializerAdapter, NetworkCallRecord record, Class<T> clazz) {
+        try {
+            return serializerAdapter.deserialize(record.getResponse().get("Body"),
+                clazz, SerializerEncoding.JSON);
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to deserialize service response.");
```

Comment:
do you want to user logger? or is that fine to throw it directly in test?

Reply:
Logger in tests?